File: CS_5530_Ass2_Trains_Project/SRC/Aggregate.r
library(dplyr)

# Read the data from the provided CSV file
data <- read.csv("/content/Clean_Modified_trains_data.csv")

# Step 1: Rename columns for better readability
data <- data %>%
  rename(Location_City = Location)

# Step 2: Mutate - Calculate the age of the car
data <- data %>%
  mutate(Car_Age = as.numeric(format(Sys.Date(), "%Y")) - Year)

# Step 3: Arrange the data by the year of manufacture (Year) in descending order
data <- data %>%
  arrange(desc(Year))

# Step 4: Summarize - Group by Location and summarize the data
summary_data <- data %>%
  group_by(Location_City) %>%
  summarize(
    Total_Cars = n(),
    Average_Kilometers = mean(Kilometers_Driven),
    Average_Price = mean(Price),
    Oldest_Car_Year = min(Year),
    Newest_Car_Year = max(Year)
  )

# View the summarized data
print(summary_data)

File: CS_5530_Ass2_Trains_Project/SRC/Aggregate.py
import pandas as pd

# Read the data from the provided CSV file
data = pd.read_csv("/content/Clean_Modified_trains_data.csv")

# Step 1: Rename columns for better readability
data.rename(columns={'Location': 'Location_City'}, inplace=True)



# Step 3: Arrange the data by the year of manufacture (Year) in descending order
data.sort_values(by='Year', ascending=False, inplace=True)

# Step 4: Summarize - Group by Location and summarize the data
summary_data = data.groupby('Location_City').agg(
    Total_Cars=('ID', 'count'),
    Average_Kilometers=('Kilometers_Driven', 'mean'),
    Average_Price=('Price', 'mean'),
    Oldest_Car_Year=('Year', 'min'),
    Newest_Car_Year=('Year', 'max')
).reset_index()

# Round the numeric columns to 2 decimal places
summary_data = summary_data.round({'Average_Kilometers': 2, 'Average_Price': 2})



df.to_csv("Sumarize_trains_data.csv", index=False)

# Display the head of the modified DataFrame

print(df.head())
